File: src/dotnet/Common/Models/Orchestration/OpenAIImageFileMessageContentItem.cs
using System.Text.Json.Serialization;
using FoundationaLLM.Common.Constants;

namespace FoundationaLLM.Common.Models.Orchestration
{
    /// <summary>
    /// An OpenAI image file message content item.
    /// </summary>
    public class OpenAIImageFileMessageContentItem : MessageContentItemBase
    {
        /// <inheritdoc/>
        [JsonIgnore]
        public override string? Type { get; set; }

        /// <summary>
        /// The ID of the image file.
        /// </summary>
        [JsonPropertyName("file_id")]
        public string? FileId { get; set; }

        /// <summary>
        /// The URL of the image file.
        /// </summary>
        [JsonPropertyName("file_url")]
        public string? FileUrl { get; set; }

        /// <summary>
        /// Set default property values.
        /// </summary>
        public OpenAIImageFileMessageContentItem() =>
            Type = MessageContentItemTypes.ImageFile;
    }
}

File: src/python/PythonSDK/foundationallm/models/orchestration/openai_image_file_message_content_item.py
from pydantic import Field
from typing import Optional
from .message_content_item_base import MessageContentItemBase
from .message_content_item_types import MessageContentItemTypes

class OpenAIImageFileMessageContentItem(MessageContentItemBase):
    """An OpenAI image file message content item."""
    
    file_id: Optional[str] = Field(None, alias="file_id")
    file_url: Optional[str] = Field(None, alias="file_url")
    
    def __init__(self, **data):
        super().__init__(**data)
        self.type = MessageContentItemTypes.IMAGE_FILE
